src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -2347,7 +2347,8 @@ void ObjectMgr::LoadItemTemplates()
         itemTemplate.ContainerSlots            = uint32(fields[26].GetUInt8());
         itemTemplate.StatsCount                = uint32(fields[27].GetUInt8());
 
-        for (uint8 i = 0; i < itemTemplate.StatsCount; ++i)
+        // Loops according to StatsCount, but never exceeds MAX_ITEM_PROTO_STATS (Client-side limited to 10)
+        for (uint8 i = 0; i < itemTemplate.StatsCount && i <= MAX_ITEM_PROTO_STATS; ++i)
```

Comment:
it's makes little sense to check i <= MAX_ITEM_PROTO_STATS and check it again later